typing/typemod.ml

```diff
@@ -1083,15 +1087,20 @@ and approx_constraint env body constr =
       let _,_,sg = Merge.merge_modtype ~approx:true ~destructive
           env smty.pmty_loc body id approx_smty in
       sg
-  (* module substitutions are ignored, but checked for cyclicity *)
-  | Pwith_module (_, lid') ->
+  (* module substitutions are approximated and merged, checking for cyclicity *)
+  | Pwith_module (id, lid)
+  | Pwith_modsubst (id, lid) ->
```

Comment:
The `Pwith_typesubst` should be handled too:
```ocaml
module Empty = struct end
module rec X: sig
  module type T = sig type t end with type t := int
  module F(A:T): sig type t end
end = struct
  module type T = sig end
  module F(A:T) = struct type t end
end
and Y: sig type t = X.F(Empty).t end = struct
   type t = X.F(Empty).t
end
```

Reply:
I think so because triggering the bug requires an interaction of non common features (application paths, recursive modules and with constraints).